ui/selectors/multichain.ts

```diff
@@ -212,9 +212,36 @@ export function getMultichainNetwork(
   // on having a non-EVM account being selected!
   const selectedAccount = account ?? getSelectedInternalAccount(state);
   const nonEvmNetworks = getMultichainNetworkProviders(state);
-  const nonEvmNetwork = nonEvmNetworks.find((provider) => {
-    return provider.isAddressCompatible(selectedAccount.address);
-  });
+  const selectedChainId = state.metamask.selectedMultichainNetworkChainId;
+
+  let nonEvmNetwork: MultichainProviderConfig | undefined;
+
+  // First try to find network by selectedChainId
+  if (selectedChainId) {
+    nonEvmNetwork = nonEvmNetworks.find(
+      (provider) => provider.chainId === selectedChainId,
+    );
+  }
+
+  // If no network found by selectedChainId, we try to find by scopes
+  if (!nonEvmNetwork && selectedAccount.scopes?.length > 0) {
+    // If we have a selectedChainId but didn't find a match, we try to find a network
+    // that matches both the selectedChainId and is in the scopes
+    if (selectedChainId) {
+      nonEvmNetwork = nonEvmNetworks.find(
+        (provider) =>
+          provider.chainId === selectedChainId &&
+          selectedAccount.scopes.includes(provider.chainId),
```

Comment:
There might be some edge-cases when comparing scopes (especially for `eip155:0`), I know this part of the code is mainly for non-EVM, but I think it'd be great to use `isScopeEqualToAny` here (as a good practice).
- https://github.com/MetaMask/accounts/blob/67b571cc3ed504ad20a5737b729ba42bd2e7fb43/packages/keyring-utils/src/scopes.ts#L49